### DISCUSSION TITLE
How to add jDeploy into an existing IntelliJ project

### DISCUSSION THREAD

**Discussion by ouaibsky** [shannah/jdeploy#119] [Q&A]
Hello

I would like to test jDeploy into my project (right now using jpackage: https://github.com/Imag-In/SPO)
I'm using IntelliJ and my project already published on GitHub.
I have installed IntelliJ - JDeploy plugin

I read the doc but I didn't see any section about this.
I create an empty package.json but I'm not able to get the menu "JDeploy settings"

Thx in advance
Christophe.

**Comment by shannah** [comment]
Right now the intellij plugin only works with projects created by the plugin (via new project dialog).

You could create a project that way and copy in your sources.

**Reply by ouaibsky** [reply]
Okay, any specific files to copy on top package.json ?

**Reply by shannah** [reply]
The github workflow.

Note that you can use the jdeploy cli also to access the same jdeploy config dialog.

npm install -g jdeploy
cd myproject
jdeploy

(Should open the GUI).

I plan to make the intellij plugin work in arbitrary projects, but need to figure out how to do that.

**Comment by charot-git** [comment]
Good day, is it possible to integrate jdeploy to an existing spring boot javafx project now?

**Reply by shannah** [reply]
A qualified "yes".  You can add jdeploy to any app.  All it needs is a package.json with a few jDeploy-specific configs.  The easiest way would be to navigate to the project directory of the javafx app, and run `jdeploy` in the cli.  It will ask if you want to generate the package.json.

You can then modify it to your needs.

The qualification here is that a spring-boot+javafx project is probably a compound project with both server-side and client-side portions, so you probably have some special application structure, and way that they should be deployed together.  I'm happy to help you get this set up for deployment.  E.g. If you have this in github, I can take a look and make a PR that sets up deployment using github actions and github releases.

**Reply by charot-git** [reply]
Hello @shannah , I got it to work! Works like a charm. The jdeploy GUI itself didnt work for me but manually editing the package.json and following the documentation in your website did the trick. It even works with jdk 21. I'll have my QA test the app out and get back to you! Thanks a lot for.

**Reply by shannah** [reply]
Glad to hear you got it working.  Can you provide more details about your experience with the jdeploy gui?  What happened when you tried?

**Reply by charot-git** [reply]
When I ran the jdeploy gui, when I pressed the publish button, it asked for my npm credentials, It did authenticate as I did see in the console., but it asked for credentials again and didnt publish the installer in the expected website.
